nifi-nar-bundles/nifi-accumulo-bundle/nifi-accumulo-services/src/main/java/org/apache/nifi/accumulo/controllerservices/AccumuloService.java

```diff
@@ -150,61 +176,92 @@ private AuthenticationToken getToken(final AuthenticationType type, final Config
+                    problems.add(new ValidationResult.Builder().valid(false).subject(KERBEROS_CREDENTIALS_SERVICE.getName())
+                            .explanation("Kerberos credential service must be supplied for the Kerberos Authentication type").build());
                 }
                 break;
             default:
-                problems.add(new ValidationResult.Builder().valid(false).subject(ACCUMULO_PASSWORD.getName()).explanation("Non supported Authentication type").build());
+                problems.add(new ValidationResult.Builder().valid(false).subject(AUTHENTICATION_TYPE.getName()).explanation("Non supported Authentication type").build());
         }
 
         return problems;
     }
 
     @OnEnabled
-    public void onEnabled(final ConfigurationContext context) throws InitializationException, IOException, InterruptedException {
-        if (!context.getProperty(INSTANCE_NAME).isSet() || !context.getProperty(ZOOKEEPER_QUORUM).isSet() || !context.getProperty(ACCUMULO_USER).isSet()){
+    public void onEnabled(final ConfigurationContext context) throws InitializationException, IOException {
+        if (!context.getProperty(INSTANCE_NAME).isSet() || !context.getProperty(ZOOKEEPER_QUORUM).isSet()) {
             throw new InitializationException("Instance name and Zookeeper Quorum must be specified");
         }
 
-
-
+        final KerberosCredentialsService kerberosService = context.getProperty(KERBEROS_CREDENTIALS_SERVICE).asControllerService(KerberosCredentialsService.class);
         final String instanceName = context.getProperty(INSTANCE_NAME).evaluateAttributeExpressions().getValue();
         final String zookeepers = context.getProperty(ZOOKEEPER_QUORUM).evaluateAttributeExpressions().getValue();
-        final String accumuloUser = context.getProperty(ACCUMULO_USER).evaluateAttributeExpressions().getValue();
+        authType = AuthenticationType.valueOf( context.getProperty(AUTHENTICATION_TYPE).getValue());
+        AuthenticationToken token;
 
-        final AuthenticationType type = AuthenticationType.valueOf( context.getProperty(AUTHENTICATION_TYPE).getValue() );
+        final Properties clientConf = new Properties();
+        clientConf.setProperty("instance.zookeepers", zookeepers);
+        clientConf.setProperty("instance.name", instanceName);
 
+        switch(authType){
+            case PASSWORD:
+                final String accumuloUser = context.getProperty(ACCUMULO_USER).evaluateAttributeExpressions().getValue();
 
+                token = new PasswordToken(context.getProperty(ACCUMULO_PASSWORD).getValue());
 
-        final AuthenticationToken token = getToken(type,context);
+                this.client = Accumulo.newClient().from(clientConf).as(accumuloUser,token).build();
+                break;
+            case KERBEROS:
+                final  String principal = kerberosService.getPrincipal();
+
+                clientConf.setProperty("sasl.enabled", "true");
```

Comment:
If you are setting these, you probably also want to set `instance.rpc.sasl.enabled=true`

Reply:
@joshelser Thanks Josh for your recommendation. My understanding is that instance.rpc.sasl.enabled is an Accumulo server property, would this be recognised by the client? I could not find it as a valid property for client configuration in Accumulo documentation ( https://accumulo..apache.org/docs/2.x/configuration/client-properties ).

Reply:
Looks like you're right. I didn't remember that they had separated client and server configuration properties. So, by virtue of passing the KerberosToken, you wouldn't need to do anything else.

Reply:
Thanks @joshelser